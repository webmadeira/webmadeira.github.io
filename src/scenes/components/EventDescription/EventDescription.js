import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import AppContainer from '../../../components/AppContainer/AppContainer'

const Container = styled.div`
  padding: 30% 10px 5% 10px;
`

const Description = styled.p`
  text-align: left;
  font-size: ${props => props.theme.textSetting('nr').fontSize};
  color: ${props => props.theme.color('neutral', 'bright')};
`

const EventDescription = ({ description, children }) => (
  <AppContainer type="primary">
    <Container>
      <Description>{description}</Description>
      {children}
    </Container>
  </AppContainer>
)

EventDescription.propTypes = {
  description: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default EventDescription
