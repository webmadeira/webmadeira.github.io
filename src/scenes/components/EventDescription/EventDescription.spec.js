import React from 'react'
import { shallow } from 'enzyme'
import EventDescription from './EventDescription'

describe('EventDescription component', () => {
  it('must render properly', () => {
    const component = shallow(
      <EventDescription description="Description sample">
        <p>Content</p>
      </EventDescription>,
    )

    expect(component).toMatchSnapshot()
  })
})
