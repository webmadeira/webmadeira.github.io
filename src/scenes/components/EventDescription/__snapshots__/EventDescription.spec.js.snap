// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDescription component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventDescription
    description="Description sample"
  >
    <p>
      Content
    </p>
  </EventDescription>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WMLogo />,
        <CtaButtons />,
        <styled.section
          type="primary"
        >
          <styled.div>
            <styled.p>
              Description sample
            </styled.p>
            <p>
              Content
            </p>
          </styled.div>
        </styled.section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <styled.p>
              Description sample
            </styled.p>
            <p>
              Content
            </p>
          </styled.div>,
          "type": "primary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.p>
                Description sample
              </styled.p>,
              <p>
                Content
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Description sample",
              },
              "ref": null,
              "rendered": "Description sample",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Content",
              },
              "ref": null,
              "rendered": "Content",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WMLogo />,
          <CtaButtons />,
          <styled.section
            type="primary"
          >
            <styled.div>
              <styled.p>
                Description sample
              </styled.p>
              <p>
                Content
              </p>
            </styled.div>
          </styled.section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.div>
              <styled.p>
                Description sample
              </styled.p>
              <p>
                Content
              </p>
            </styled.div>,
            "type": "primary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.p>
                  Description sample
                </styled.p>,
                <p>
                  Content
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Description sample",
                },
                "ref": null,
                "rendered": "Description sample",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Content",
                },
                "ref": null,
                "rendered": "Content",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
